assets/js/base/utils/render-frontend.js

```diff
@@ -53,4 +79,27 @@ export const renderFrontend = ( {
 	}
 };
 
+/**
+ * Adds the event listeners necessary to render the block frontend.
+ *
+ * @param {Object} props Render props.
+ */
+export const renderFrontend = ( props ) => {
+	const wrappersToSkipOnLoad = document.body.querySelectorAll(
+		selectorsToSkipOnLoad.join( ',' )
+	);
+	// Render on page load.
+	renderBlockFrontend( {
+		...props,
+		wrappersToSkip: wrappersToSkipOnLoad,
+	} );
+	// Render wrappers inner blocks when the event `wc-blocks_render_blocks_frontend`
+	// is triggered.
+	Array.prototype.forEach.call( wrappersToSkipOnLoad, ( wrapper ) => {
+		wrapper.addEventListener( 'wc-blocks_render_blocks_frontend', ( e ) => {
+			renderBlockFrontend( { ...props, wrapper: e.target } );
+		} );
+	} );
+};
+
```

Comment:
While I get the higher-level logic, I'm having a hard time understanding the logic happening here, what exactly are we doing? waiting for an event to render again?

Reply:
`renderFrontend` does three things:
* Gets the wrapper elements that should be skipped on page load. That means that inner blocks of those wrappers should not be rendered on page load but instead when an event on the wrapper is triggered.
* Renders the block frontend for all blocks which are not inside those wrappers.
* For each of those wrappers, adds an event listener so, when the `wc-blocks_render_blocks_frontend` event is triggered, inner blocks are rendered.

In a343fc08c6d02ac9476ba6131eb8f8a3e03fb7c5, I spent some time renaming functions, variables and re-organizing code trying to make code a bit simpler. Hopefully it's a bit easier to understand now. :slightly_smiling_face: